### PR TITLE
Improve dropdowns

### PR DESCRIPTION
**Pull request by jacobjzhang**
Associated Issue: Jason Laster Recommendation

### Summary of Changes

* Extrapolated tab dropdown as a dropdown component
* Used it for excess tabs when files are open

### Testing

* [X] passes `npm test`
* [X] passes `npm run lint`

### CODE REVIEW

**Comment by wldcordeiro** [comment]
Two commits should be removed from this PR.
`197ced35d679ff93242b8f37bd39e62cd180f45f` and `a5ff99f5f5ca9d80943885fa2c00e58a8b514b71` as they're unrelated to the dropdowns and in the first one's case redo work that has already been implemented.

**Comment by wldcordeiro** [comment]
@jacobjzhang Changes requested, make sure to read through issues and keep PRs topical to the issue you're fixing, since this has multiple things that weren't associated with dropdowns in it. Also make a habit of rebasing/merging master with your fork prior to making PRs. Thanks for the work though, just need to keep things ordered and not retread ground already covered. 😄

File: public/js/components/Dropdown.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  toggleSourcesDropdown(e) {
+    this.setState({
+      dropdownShown: !this.state.dropdownShown,
+    });
+  },
+
+  renderPanel() {
+    return dom.div(
+      {
+        className: "sources-dropdown dropdown",
```
Comment (jlongster) [inline public/js/components/Dropdown.js]: Please don't use the term "sources" in a generic that is not tied to sources at all. If this is supposed to be a generic dropdown component, just use generic terms.

File: public/js/components/SourceTabs.css [hunk 2]
```diff
@@ -63,3 +63,17 @@
   top: 35px;
   right: 0px;
 }
+
+.sources-dropdown li {
```
Comment (jlongster) [inline public/js/components/SourceTabs.css]: The dropdown component should have its own CSS file.

**Comment by jlongster** [comment]
@jacobjzhang Are you willing to pick this up again? If not, we should close this PR.

**Comment by jacobjzhang** [comment]
@jlongster Sorry for delay and thanks for feedback, picking this up again tonight after @jasonLaster helped me with rebasing.

**Comment by jacobjzhang** [comment]
Ran into some rebasing issues. Going to re-fork off the main repo and re-extract out the dropdown.

**Pull request closed without merging**